spec/integration_spec.rb

```diff
@@ -0,0 +1,57 @@
+# encoding: utf-8
+require "lib/lumberjack/client"
+require "lib/lumberjack/server"
+require "stud/temporary"
+require "flores/pki"
+require "fileutils"
+require "thread"
+require "spec_helper"
+
+describe "A client" do
+  let(:port) { Flores::Random.integer(1024..65335) }
+  let(:csr) { Flores::PKI::CertificateSigningRequest.new }
+  let(:key_bits) { 1024 }
+  let(:key) { OpenSSL::PKey::RSA.generate(key_bits, 65537) }
+  let(:certificate_duration) { Flores::Random.number(1..86400) }
+  let(:certificate_subject) { "CN=server.example.com" }
+  let(:certificate) { csr.create }
+  let(:certificate_file_crt) { "certificate.crt" }
+  let(:certificate_file_key) { "certificate.key" }
+  let(:host) { "127.0.0.1" }
+  let(:queue) { Queue.new }
+  let(:payload) { {"line" => "hello world" } }
+
+  before do
+    csr.subject = certificate_subject
+    csr.public_key = key.public_key
+    csr.start_time = Time.now
+    csr.expire_time = csr.start_time + certificate_duration
+    csr.signing_key = key
+    csr.want_signature_ability = true
+
+    expect(File).to receive(:read).twice.with(certificate_file_crt) { certificate.to_s }
+    expect(File).to receive(:read).with(certificate_file_key) { key.to_s }
+    
+    server = Lumberjack::Server.new(:port => port,
+                                    :address => host,
+                                    :ssl_certificate => certificate_file_crt,
+                                    :ssl_key => certificate_file_key)
+
+    Thread.new do
+      server.run { |data| queue.push(data) }
+    end
+  end
+
+  it "should require a certificate" do
+    expect { 
+      client = Lumberjack::Client.new(:port => port, 
+                             :host => host,
+                             :addresses => host,
+                             :ssl_certificate => certificate_file_crt)
```

Comment:
Well the client use the public key to encrypt which is available in the certificate and the server use the private key to decrypt it? The ssl_key was never a requirement in the `logstash-output-lumberjack`. But yes the LSF has an optional key option in the config, I'll check it more.